src/kernel/tty.zig

```diff
@@ -361,55 +387,58 @@ fn writeString(str: []const u8) PrintError!void {
-///
-/// Arguments:
-///     IN char: u8 - The character to print.
+/// Print the pluto logo.
 ///
 fn printLogo() void {
     const column_temp = column;
     const row_temp = row;
 
+    defer column = column_temp;
+    defer row = row_temp;
+
+    const logo1 = "                  _____    _        _    _   _______    ____  \n";
+    const logo2 = "                 |  __ \\  | |      | |  | | |__   __|  / __ \\ \n";
+    const logo3 = "                 | |__) | | |      | |  | |    | |    | |  | |\n";
+    const logo4 = "                 |  ___/  | |      | |  | |    | |    | |  | |\n";
+    const logo5 = "                 | |      | |____  | |__| |    | |    | |__| |\n";
+    const logo6 = "                 |_|      |______|  \\____/     |_|     \\____/ \n";
+
+    // Print the logo at the top of the screen
     column = 0;
     row = 0;
 
-    writeString("                  _____    _        _    _   _______    ____  \n") catch unreachable;
-    writeString("                 |  __ \\  | |      | |  | | |__   __|  / __ \\ \n") catch unreachable;
-    writeString("                 | |__) | | |      | |  | |    | |    | |  | |\n") catch unreachable;
-    writeString("                 |  ___/  | |      | |  | |    | |    | |  | |\n") catch unreachable;
-    writeString("                 | |      | |____  | |__| |    | |    | |__| |\n") catch unreachable;
-    writeString("                 |_|      |______|  \\____/     |_|     \\____/ \n") catch unreachable;
+    writeString(logo1 ++ logo2 ++ logo3 ++ logo4 ++ logo5 ++ logo6) catch |_| {
+        log.logError("TTY: Error print logo\n");
+    };
+}
 
-    column = column_temp;
-    row = row_temp;
+///
+/// A call back function for use in the formation of a string. This calls writeString normally.
+///
+/// Arguments:
+///     IN ctx: void       - The context of the printing. This will be empty.
+///     IN str: []const u8 - The string to print.
+///
+/// Errors:
+///     TtyError.OutOfBounds - If trying to print beyond the video buffer.
+///
+fn printCallback(ctx: void, str: []const u8) TtyError!void {
+    try writeString(str);
 }
 
 ///
 /// Print a formatted string to the terminal in the current colour. This used the standard zig
 /// formatting.
 ///
 /// Arguments:
-///     IN format: []const u8 - The format string to print
-///     IN args: ...          - The arguments to be used in the formatted string
+///     IN comptime format: []const u8 - The format string to print
+///     IN args: ...                   - The arguments to be used in the formatted string
 ///
 pub fn print(comptime format: []const u8, args: ...) void {
     // Printing can't error because of the scrolling, if it does, we have a big problem
-    fmt.format({}, PrintError, printCallback, format, args) catch unreachable;
+    fmt.format({}, TtyError, printCallback, format, args) catch |_| {
+        log.logError("TTY: Error printing\n");
```

Comment:
Would be nice to print the error name with `@errorName`. Same with a few more places in this file.

Reply:
I was thinking of this, will add printing the error